crates/bevy_image/src/image.rs

```diff
@@ -878,6 +880,56 @@ impl Image {
         self.texture_descriptor.size = new_size;
     }
 
+    /// Resizes the image to the new size, keeping the pixel data intact, anchored at the top-left.
+    /// When growing, the new space is filled with `fill`. When shrinking, the image is clipped.
+    ///
+    /// For faster resizing when keeping pixel data intact is not important, use [`Image::resize`].
+    pub fn resize_in_place_2d(
+        &mut self,
+        new_size: Extent3d,
```

Comment:
Maybe `new_size` should be a `UVec2`, it feels odd to ask for a depth value unless it's used for something?